src/bin/main.rs

```diff
@@ -22,11 +29,29 @@ async fn main() {
         .version(show_version(crate_info).as_str())
         .author("Dusk Network B.V. All Rights Reserved.")
         .about("Rusk Server node.")
+        .arg(
+            Arg::with_name("socket")
+                .short("s")
+                .long("socket")
+                .value_name("socket")
+                .help("Path for setting up the UDS ")
+                .default_value(SOCKET_PATH)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name("ipc_method")
+                .long("ipc_method")
+                .value_name("ipc_method")
+                .possible_values(&["uds", "tcp_ip"])
```

Comment:
`uds` isn't possible with windows. Maybe the provided `possible_values` will change depending on the target os?

Reply:
I tried to make a cgf_attrib. But makes the code less readable.
Therefore, I accept on the arg the two options and parse later depending on the `target_os` the options choosen.